birdhouse/components/stac/config/magpie/config.yml.template

```diff
@@ -0,0 +1,19 @@
+providers:
+  # definition of STAC service for API access
+  stac:
+    url: http://stac:8000
+    title: STAC
+    public: true
+    c4i: false
+    type: api
+    sync_type: api
+
+permissions:
+  - service: stac
+    permission: read
+    group: anonymous
```

Comment:
Is there a reason why this is added here instead of in the `all-public-access` optional component? 

For almost every other service, the default is that it is not accessible through magpie by default unless `all-public-access` is enabled.

Reply:
I understand. My point is that we don't assign anonymous permissions by default for any other component. Is there a reason why stac is different from the other components and needs to have the permissions handled differently?

Reply:
I thought the idea was that the person administering the node chooses what permissions to enable (by logging in to magpie). 

For example, if you're using finch in your stack right now (without `all-public-access`) then users can't query the `getcapabilities` from finch out of the box, the permissions have to be set.

If this is not the pattern we want to follow with stac, that's fine. I'm just pointing out that it's different from what other components are doing right now.

Reply:
I tend to agree with @mishaschwartz (*somewhat* 😅)

The node administrator should indeed add a custom Magpie `config.yml` or `permissions.cfg` that applies any relevant permissions to match their needs for their own platform.

Adding this permission to https://github.com/bird-house/birdhouse-deploy/tree/master/birdhouse/optional-components/all-public-access will document on how to do this specifically for the STAC service.  It doesn't mean we rely on it necessarily, since each platform has to choose to opt in to the `stac` component anyway.

However, considering that STAC (specifically the `stac-browser`) is eventually intended to be an alternative to JupyterHub as server entrypoint, it could be considered to have an alternate optional component `stac-browser-entrypoint` specifically for setting this up. In that component, it would make sense to have the public permission applied automatically, since the server would fail immediately if accessed publicly otherwise. 

Also, given that STAC is essentially only for metadata search, I am not that much worried if the permission was left there either. Similar public setup is assumed and defined for other components, (e.g.: https://github.com/bird-house/birdhouse-deploy/blob/master/birdhouse/components/weaver/config/magpie/config.yml.template). Therefore, this rule of custom permissions is not *strictly* applied for all components.

Reply:
Yes, but add it as a separate `optional-components/stac-public-access`. 
I can see STAC being made public by itself more often while keeping the rest protected.

Also, `optional-components/all-public-access` would depend on `optional-components/stac-public-access`.